research/blanks/packer.py

```diff
@@ -37,13 +38,21 @@ def __init__(
 
         assert len(self.blanks_ids) == self.n_blanks
         self.use_only_last_blank_loss = use_only_last_blank_loss
+        self.insert_blanks_without_replacement = insert_blanks_without_replacement
 
     def get_sample(self) -> BlanxExample:
         sample = super().get_sample()
 
         input_tokens = sample.input_ids
         target_tokens = sample.target_ids
         seq_len = len(input_tokens)
+        if self.insert_blanks_without_replacement:
+            input_tokens.extend(
+                [0] * self.n_blanks
+            )  # will be cut off by inserting blanks
+            target_tokens.extend([0] * self.n_blanks)
+            seq_len += self.n_blanks
+            sample.should_calculate_loss.extend([1] * self.n_blanks)
```

Comment:
Maybe `sample.should_calculate_loss.extend([-1] * self.n_blanks)` and put an assert at the end of the function checking that there is no -1's.

Reply:
Not how it works, this is either copied without any change or made from grouds up. We already have an assert should_calculate_loss == [1] * seq_len